pyats_installer.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/python
+import subprocess
+import argparse
+import pkg_resources
+from distutils.version import StrictVersion
+LATEST = '19.12'
+
+VERSION_MAPPING = {
+    '20.1': {
+        'install': ['pyats[full]'],
+        'uninstall': ['pyats[full]']
+    },
+    '19.11':{
+        'install': ['unicon', 'unicon.plugins'],
+        'uninstall': ['unicon']
+    }
+}
+# todo find a way to automatically find pkg deps+extras
+# pkg_resource.require doesnt work b/c it raise exception if the dep is not installed
+PYATS_PKG_DEPENDENCIES = {
+    'pyats',
+    'pyats.aereport',
+    'pyats.aetest',
+    'pyats.async',
+    'pyats.cisco',
+    'pyats.log',
+    'pyats.kleenex',
+    'pyats.connections',
+    'pyats.datastructures',
+    'pyats.easypy',
+    'pyats.results',
+    'pyats.reporter',
+    'pyats.tcl',
+    'pyats.topology',
+    'pyats.utils'}
+
+# pre 20.1
+OLD_GENIE_PKG_DEPENDENCIES = {
+    'genie',
+    'genie.abstract',
+    'genie.metaparser',
+    'genie.libs.parser',
+    'genie.parsergen',
+    'genie.telemetry',
+    'genie.libs.telemetry',
+    'genie.conf',
+    'genie.ops',
+    'genie.harness',
+    'genie.predcore',
+    'genie.utils',
+    'genie.libs.conf',
+    'genie.libs.ops',
+    'genie.libs.sdk',
+    'genie.libs.filetransferutils',
+    'genie.trafficgen'}
+
+# after 20.1
+NEW_GENIE_PKG_DEPENDENCIES = {
+    'genie',
+    'genie.libs.conf',
+    'genie.libs.filetransferutils',
+    'genie.libs.ops',
+    'genie.libs.parser',
+    'genie.libs.sdk',
+    'genie.telemetry',
+    'genie.trafficgen'}
+
+ROBOT = {
+    'pyats.robot',
+    'genie.libs.robot'
+}
+
+UNICON = {
+    'unicon', 'unicon.plugins'
+}
+
+
+class PyatsInstaller:
+
+    def __init__(self, extra, version=None, uninstall=None, **kwargs):
+        self.uninstall = uninstall
+        self.extra = extra
+        self.version = version or LATEST
+        # genie deps changed after 20.1
+        self.GENIE_PKG_DEPENDENCIES = NEW_GENIE_PKG_DEPENDENCIES if StrictVersion(
+            self.version) >= StrictVersion('20.1') else OLD_GENIE_PKG_DEPENDENCIES
+        
+        try:
+            import pyats.aetest
```

Comment:
i think the best way to go about this is to run a pip uninstall command.

Reply:
I'm not sure about how pip uninstall command could be used here....